builder-base/Dockerfile

```diff
@@ -12,20 +12,141 @@
+RUN /install_buildkit.sh
+
+FROM downloader as ecr-cred-helper
+COPY ./scripts/install_ecr_cred_helper.sh ./scripts/common_vars.sh \
+    ./checksums/amazon-ecr-cred-helper-amd64-checksum ./checksums/amazon-ecr-cred-helper-arm64-checksum /
+RUN /install_ecr_cred_helper.sh
+
+FROM downloader as gh_cli
+COPY ./scripts/install_gh_cli.sh ./scripts/common_vars.sh \
+    ./checksums/github-cli-amd64-checksum /
+RUN /install_gh_cli.sh
+
+FROM downloader as docker_buildx
+COPY ./scripts/install_docker_buildx.sh ./scripts/common_vars.sh \
+    ./checksums/docker-buildx-amd64-checksum /
+RUN /install_docker_buildx.sh
+
+FROM downloader as yq
+COPY ./scripts/install_yq.sh ./scripts/common_vars.sh \
+    ./checksums/yq-amd64-checksum ./checksums/yq-arm64-checksum /
+RUN /install_yq.sh
+
+FROM downloader as packer
+COPY ./scripts/install_packer.sh ./scripts/common_vars.sh \
+    ./checksums/packer-amd64-checksum /
+RUN /install_packer.sh
 
+FROM downloader as helm
+COPY ./scripts/install_helm.sh ./scripts/common_vars.sh \
+    ./checksums/helm-amd64-checksum /
+RUN /install_helm.sh
+
+FROM downloader as nodejs
+COPY ./scripts/install_nodejs.sh ./scripts/common_vars.sh \
+    generate-attribution/package*.json  \
+    generate-attribution/generate-attribution  \
+    generate-attribution/generate-attribution-file.js \ 
+    generate-attribution/LICENSE-2.0.txt \
+    ./checksums/nodejs-amd64-checksum /
+RUN /install_nodejs.sh
+
+FROM downloader as hugo
+COPY ./scripts/install_hugo.sh ./scripts/common_vars.sh \
+    ./checksums/hugo-amd64-checksum /
+RUN /install_hugo.sh
+
+FROM downloader as govc
+COPY ./scripts/install_govc.sh ./scripts/common_vars.sh \
+    ./checksums/govc-amd64-checksum /
+RUN /install_govc.sh
+
+FROM downloader as goss
+COPY ./scripts/install_goss.sh ./scripts/common_vars.sh \
+    ./checksums/goss-amd64-checksum /
+RUN /install_goss.sh
+
+FROM downloader as skopeo
 ARG GOPROXY
-ARG TARGETARCH
+COPY ./scripts/install_skopeo.sh ./scripts/common_vars.sh /
+RUN export GOPROXY=${GOPROXY} && \
+    /install_skopeo.sh
 
-ENV GOPATH /go
+FROM downloader as tuftool
 ENV CARGO_HOME /root/.cargo
 ENV RUSTUP_HOME /root/.rustup
-ENV PATH="/go/bin/:/root/.cargo/bin:$PATH"
+ENV PATH="/root/.cargo/bin:$PATH"
+COPY ./scripts/install_tuftool.sh ./scripts/common_vars.sh /
+RUN /install_tuftool.sh
+
+FROM downloader as bash
+COPY ./scripts/install_bash.sh ./scripts/common_vars.sh \
+    ./checksums/bash-checksum /
+RUN /install_bash.sh
+
+FROM downloader as ansible
+COPY ./scripts/install_ansible.sh ./scripts/common_vars.sh /
+RUN /install_ansible.sh
+
+FROM downloader as golang
+ARG GOPROXY
+ENV GOPATH /go
+ENV PATH="/go/bin/:$PATH"
+COPY ./scripts/install_golang.sh ./scripts/common_vars.sh \
+    ./checksums/golang-amd64-checksum ./checksums/golang-arm64-checksum /
+RUN export GOPROXY=${GOPROXY} && \
+    /install_golang.sh
+
+
+FROM ${BASE_IMAGE} as copy-stage
+
+# Ordered by least likely to be updated
```

Comment:
This is fairly minor, probably only affects those of us who pull the builder base locally, but im open to opinions on the ordering here.  Since these are copy --link layers they do not depend on each other so cache can be reused between builds, however, the final layers would end up being different if any of the previous layers change, because the final layer has the parent layer hash in it.  If we make sure and put the layers that change most often towards the end then we can reuse the final layers more and more between builds.

Reply:
that sounds good to me. We know we'll be updating Go pretty frequently, so having that toward the front of the list makes sense. We can prolly revisit the order as needed if something should be moved up or down the list